app/v1/services/NrsService.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import java.lang.String.format
+import java.math.BigInteger
+import java.nio.charset.StandardCharsets.UTF_8
+import java.security.MessageDigest.getInstance
+import v1.models.requests.IdentifierRequest
+import javax.inject.{Inject, Singleton}
+import com.google.common.io.BaseEncoding.base64
+import play.api.libs.json.{JsObject, JsString, JsValue}
+import v1.connectors.NrsConnector
+import uk.gov.hmrc.http._
+import play.api.Logging
+import v1.models.nrs._
+import v1.connectors.HttpHelper.NrsResponse
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class NrsService @Inject()(nrsConnector: NrsConnector, dateTimeService: DateTimeService)(implicit ec: ExecutionContext) extends HttpErrorFunctions with Logging {
+
+  private val searchKey = "searchKey" //TODO: Update once NRS have updated the Calling Service Registry with a value for us https://confluence.tools.tax.service.gov.uk/display/NR/Calling+Service+Registry
+  private val applicationJson = "application/json"
+  private val businessIdValue = "irr" //TODO: Update once NRS have updated the Calling Service Registry with a value for us https://confluence.tools.tax.service.gov.uk/display/NR/Calling+Service+Registry
+  private val notableEventValue = "irr-submission" //TODO: Update once NRS have updated the Calling Service Registry with a value for us https://confluence.tools.tax.service.gov.uk/display/NR/Calling+Service+Registry
```

Comment:
Do we want to put these in config?

Reply:
For now I want to leave them in here until we know what format they're going to take. I imagine in the end they'll be static values for our API and so could be moved to a constants file rather than config.